ports/raspberrypi/common-hal/watchdog/WatchDogTimer.c

```diff
@@ -37,47 +37,50 @@ void common_hal_watchdog_feed(watchdog_watchdogtimer_obj_t *self) {
 }
 
 void common_hal_watchdog_deinit(watchdog_watchdogtimer_obj_t *self) {
-    if (self->mode == WATCHDOGMODE_RESET) {
-        mp_raise_RuntimeError(translate("WatchDogTimer cannot be deinitialized once mode is set to RESET"));
-    } else {
-        self->mode = WATCHDOGMODE_NONE;
+    if (self->mode == WATCHDOGMODE_NONE) {
+        return;
```

Comment:
this test is unneeded, unless it's not safe to clear the watchdog flag when it's already cleared.